src/main/java/org/spongepowered/api/data/type/RailDirection.java

```diff
@@ -24,11 +24,103 @@
  */
 package org.spongepowered.api.data.type;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
 import org.spongepowered.api.CatalogType;
+import org.spongepowered.api.Sponge;
 import org.spongepowered.api.util.Cycleable;
+import org.spongepowered.api.util.Direction;
 import org.spongepowered.api.util.annotation.CatalogedBy;
 
+import java.util.Optional;
+
 @CatalogedBy(RailDirections.class)
 public interface RailDirection extends CatalogType, Cycleable<RailDirection> {
 
+    /**
+     * Returns the {@link RailDirection} from the given facing directions.
+     *
+     * @param firstDirection The first direction
+     * @param secondDirection The second direction
+     * @return The {@link RailDirection}
+     * @throws IllegalArgumentException If the given directions don't differ
+     */
+    static RailDirection fromFaces(final Direction firstDirection, final Direction secondDirection) {
+        checkNotNull(firstDirection, "firstDirection");
+        checkNotNull(secondDirection, "secondDirection");
+
+        Direction cardinalFirstDirection = Direction.getClosest(firstDirection.asOffset(), Direction.Division.CARDINAL);
+        Direction cardinalSecondDirection = Direction.getClosest(secondDirection.asOffset(), Direction.Division.CARDINAL);
+
+        if (cardinalFirstDirection.equals(cardinalSecondDirection)) {
+            throw new IllegalArgumentException("The directions should be different!");
+        }
+
+        for (RailDirection currentRailDirection: Sponge.getRegistry().getAllForMinecraft(RailDirection.class)) {
+            if (currentRailDirection.isFacing(cardinalFirstDirection) && currentRailDirection.isFacing(cardinalSecondDirection)) {
+                return currentRailDirection;
+            }
+        }
+
+        throw new AssertionError();
```

Comment:
No, never throw an `AssertionError` - this should be a `IllegalArgumentException`.